function import `reflink` is private because fn reflink_not_supported is not public

I guess the intention was that on a unsupported OS such as illumos this crate should also work but return an error.

Unfortunately thats broken because the function is not public.

```
error[E0603]: function import `reflink` is private
  --> /var/lib/abuild/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reflink-copy-0.1.12/src/sys/mod.rs:10:29
   |
10 |         pub use self::unix::reflink;
   |                             ^^^^^^^ private function import
   |
note: the function import `reflink` is defined here...
  --> /var/lib/abuild/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reflink-copy-0.1.12/src/sys/unix/mod.rs:11:13
   |
11 |         use super::reflink_not_supported as reflink;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: ...and refers to the function `reflink_not_supported` which is defined here
  --> /var/lib/abuild/.cargo/registry/src/index.crates.io-6f17d22bba15001f/reflink-copy-0.1.12/src/sys/mod.rs:20:1
   |
20 | fn reflink_not_supported(_from: &Path, _to: &Path) -> std::io::Result<()> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0603`.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -131,3 +131,24 @@ jobs:
     - uses: Swatinem/rust-cache@v2
 
     - run: cargo check --all-features
+
+  cross-check-unsupported-platform:
+    strategy:
+      fail-fast: false
+      matrix:
+        target: [x86_64-unknown-illumos]
+
+    env:
+      CARGO_BUILD_TARGET: ${{ matrix.target }}
+
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install rustup target
+      run: |
+        rustup toolchain install --profile minimal nightly
+
+    - uses: Swatinem/rust-cache@v2
+
+    - name: Cross check
+      run: cargo +nightly check --target ${{ matrix.target }} -Zbuild-std

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -146,7 +146,7 @@ jobs:
     - uses: actions/checkout@v4
     - name: Install rustup target
       run: |
-        rustup toolchain install --profile minimal nightly
+        rustup toolchain install --profile minimal --component rust-src nightly
 
     - uses: Swatinem/rust-cache@v2
 

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -146,7 +146,7 @@ jobs:
     - uses: actions/checkout@v4
     - name: Install rustup target
       run: |
-        rustup toolchain install --profile minimal --component rust-src nightly
+        rustup toolchain install nightly --profile minimal --component rust-src
 
     - uses: Swatinem/rust-cache@v2
 

--- a/src/sys/mod.rs
+++ b/src/sys/mod.rs
@@ -12,11 +12,11 @@ cfg_if! {
         mod windows_impl;
         pub use self::windows_impl::reflink;
     } else {
-        use self::reflink_not_supported as reflink;
+        pub use self::reflink_not_supported as reflink;
     }
 }
 
 #[allow(dead_code)]
-fn reflink_not_supported(_from: &Path, _to: &Path) -> std::io::Result<()> {
+pub fn reflink_not_supported(_from: &Path, _to: &Path) -> std::io::Result<()> {
     Err(std::io::ErrorKind::Unsupported.into())
 }

--- a/src/sys/unix/mod.rs
+++ b/src/sys/unix/mod.rs
@@ -8,6 +8,6 @@ cfg_if! {
         mod macos;
         pub use macos::reflink;
     } else {
-        use super::reflink_not_supported as reflink;
+        pub use super::reflink_not_supported as reflink;
     }
 }

